src/deadline/job_attachments/models.py

```diff
@@ -390,8 +391,65 @@ class GlobConfig:
 class ManifestSnapshot:
     """Data structure to store the results of a manifest snapshot"""
 
+    root: str = field(default_factory=str)
     manifest: str = field(default_factory=str)
 
+    @classmethod
+    def from_dict(cls, data: dict) -> "ManifestSnapshot":
+        """
+        Create a ManifestSnapshot from a dictionary.
+
+        Args:
+            data: Dictionary containing root and manifest fields
+
+        Returns:
+            ManifestSnapshot object
+
+        Raises:
+            ValueError: If the dictionary is missing required fields
+        """
+        if "root" not in data or "manifest" not in data:
+            raise ValueError("Invalid dictionary: missing required fields 'root' or 'manifest'")
```

Comment:
Instead of having dedicated serialization functionality here (since we're really just wrapping built-in dataclasses/json functions), how about we use dataclasses `__post_init__` instead? It looks like we are mainly wanting to add this validation to the class.

https://docs.python.org/3/library/dataclasses.html#post-init-processing